### PR TITLE
Add table of contents. Fixes #85

### LINKED ISSUE #85: Support TabelOfContent
I really like your hugo website design. It's simple and beautiful. Well, this is just an advice. I am wondering whether it can add the TOC supporting so that there is a catalogue displayed in the side when we read the blog?

### PR DESCRIPTION
**Pull request by 526avijitgupta**
- [x] Create ToC dynamically
- [x] Automatically highlight current position in ToC
- [x] Do not show ToC bar in smaller screens
- [x] Smooth scroll to a heading when clicked from ToC bar
- [x] Change headings heading in markdown syntax guide to H1
- [x] Move Box to a bit more top and right
- [x] ~~Experiment with adding a line in b/w content and ToC~~ (looks bad)
- [x] ~~Update URL with anchor tag~~ (as someone had guessed.. smooth scrolling does not work if you update URL)
- [x] Update documentation and README
- [x] Should be optional via attributes
- [x] If Table of contents is disabled, content should be rendered centrally
- [x] Make sure indentation for h1,h2,h3 and h4 tags work correctly

Fixes #85

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 .DS_Store
 geckodriver.log
+.idea/
```
Comment (yashmehrotra) [inline .gitignore:3]: 🤮

File: layouts/partials/post.html [hunk 2]
```diff
@@ -20,4 +20,10 @@ <h1>{{ .Title }}</h1>
         </p>
     </div>
 </div>
+<aside class="post-toc">
+    <nav id="toc">
+        <ul class="post-toc-content"></ul>
+    </nav>
+</aside>
```
Comment (yashmehrotra) [inline layouts/partials/post.html:27]: Remove extra lines below

Comment (yashmehrotra) [reply layouts/partials/post.html:27]: Lets make this a partial and include it in both post and page

Add if/else attribute to enable

File: static/css/main.css [hunk 3]
```diff
@@ -52,6 +53,75 @@ main#content {
   width: var(--content-width);
 }
 
+main#content > .container {
+  margin: auto;
+  width: 74%;
```
Comment (yashmehrotra) [inline static/css/main.css:58]: Magic number ?

File: static/js/main.js [hunk 4]
```diff
@@ -36,6 +43,74 @@ window.addEventListener('scroll', () => {
...
+      //append parent class to heading
+      $heading.classList.add('anchor-heading');
+
+      //create anchor
+      $anchor = document.createElement('a');
+      $anchor.className = 'anchor-link';
+      $anchor.href = '#' + id;
+      $anchor.innerText = $heading.innerText;
+
+      //create li element
```
Comment (yashmehrotra) [inline static/js/main.js:65]: Make this map

File: static/js/main.js [hunk 5]
```diff
@@ -36,6 +43,74 @@ window.addEventListener('scroll', () => {
...
+          break;
+        case 'H2':
+          $li.className = 'level-2';
+          break;
+        case 'H3':
+          $li.className = 'level-3';
+          break;
+        case 'H4':
+          $li.className = 'level-4';
+          break;
```
Comment (yashmehrotra) [inline static/js/main.js:80]: Make this map

Comment (yashmehrotra) [reply static/js/main.js:80]:
```
var = {'H1': 'level-1', 'H2': 'level-2' ...}
$li.className = var[$heading.tagName]
```

**Comment by yashmehrotra** [comment]
- [ ] Change headings heading in markdown syntax guide to H1
- [ ] Move Box to a bit more top and right
- [ ] Experiment with adding a line in b/w content and ToC

File: .gitignore (hunk 1 above)
Comment (yashmehrotra) [reply .gitignore:3]: Add .hugo_build.lock to .gitignore

File: static/css/main.css (hunk 3 above)
Comment (yashmehrotra) [reply static/css/main.css:58]: Add comment for reasoning

File: static/css/main.css [hunk 6]
```diff
@@ -52,6 +53,75 @@ main#content {
...
+  box-sizing: border-box;
+  display: inline-block;
+  padding-left: 20px;
+}
+
+.post-toc {
+  display: inline-block;
+  margin-left: 25px;
+  vertical-align: top;
+  width: 23%;
```
Comment (yashmehrotra) [inline static/css/main.css:72]: Add reasoning and association with other variable

**Comment by 526avijitgupta** [comment]
@yashmehrotra Please update documentation and make optional. Then merge if everything fine

**Pull request merged**